variables.tf

```diff
@@ -187,3 +187,27 @@ variable "weave_dd_profiling_enabled" {
   description = "Whether to enable Datadog profiling for Weave."
   default     = false
 }
+
+variable "weave_storage_class" {
+  type        = string
+  description = "The storage class to use for the weave cache."
+  default     = "standard"
+}
+
+variable "weave_storage_size" {
+  type        = string
+  description = "The size of the weave cache, we don't mount a pvc when not set."
```

Comment:
When we execute conditionals, rather than evaluating a value here for a boolean outcome (make the claim/don't make the claim), use a dedicated variable. In other words, create a "make_pvc_claim" variable (or something similar) to enable this functionality, as opposed to using the same variable that determines the size of the claim. 

This requires we explicitly indicate whether or not to make the claim, we avoid complex evaluation statements, and we can use more appropriate types (like number, here).